File: Modulo4-ObjectiveC-Lorena/Exercicio4/Exercicio4.m
//
//  main.m
//  Exercicio4-C
//
//  Created by Lorena Rodrigues Bruno on 10/09/2018.
//  Copyright © 2018 Lorena Rodrigues Bruno. All rights reserved.
//

#import <Foundation/Foundation.h>
#import "Contador.h"

@implementation Contador
@synthesize valorAtual, valorInicial;
-(id)initValor : (int) valor {
    self = [super init];
    valorInicial = valor;
    valorAtual = valorInicial;
    return self;
}


-(int)incrementa:(int)incremento {
    return valorAtual += incremento;
}

-(int)decrementa:(int)decremento {
    return valorAtual -= decremento;
}

-(int)informaValorInicial {
    return valorInicial;
}
-(void)imprimeValorAtual {
    printf("Valor Atual: %i\n", valorAtual);
}

@end

int main(int argc, const char * argv[]) {
    @autoreleasepool {
        Contador *contador = [[Contador alloc] initValor: 15];
        [contador incrementa:10];
        [contador decrementa:7];
        printf("Valor Inicial: %i\n", [contador valorInicial]);
        [contador imprimeValorAtual];
        
    }
    return 0;
}

File: Modulo4-Lorena/Exercicio4.swift
//
//  main.swift
//  Exercicio4
//
//  Created by Lorena Rodrigues Bruno on 28/08/2018.
//  Copyright © 2018 Lorena Rodrigues Bruno. All rights reserved.
//

import Foundation

class Contador {
    var valorAtual = Int()
    
    init(valorAtual : Int) {
        self.valorAtual = valorAtual
    }
    
    func incrementaValorAtual (incremento : Int) {
        valorAtual += incremento
    }
    
    func decrementaValorAtual (decremento : Int) {
        valorAtual -= decremento
    }
    
    func imprimeValorAtual(){
        print("Valor atual: \(valorAtual)")
    }
}

//teste:
//var contador = Contador(valorAtual: 15)
//contador.incrementaValorAtual(incremento: 10)
//contador.decrementaValorAtual(decremento: 7)
//contador.imprimeValorAtual()


